src/test/crudella.test.ts

```diff
@@ -191,4 +192,31 @@ describe('createService', () => {
             expect(methods.detail.mock.calls[0][0].options).toMatchSnapshot();
         });
     });
+    describe('Builder', () => {
+        test('Setting http context via builder', async () => {
+            type DummyContext = [number, string];
+            const serviceFactory = buildService<PersonAttributes, DummyContext>({}).createService;
+            const service = serviceFactory({
+                detail: methods.detail,
+                authorize: ctx => {
+                    const tryAssignContext: DummyContext = ctx.context;
+                    return tryAssignContext;
+                },
+            });
+        });
+        test('Composition build', async () => {
+            const serviceFactory = buildService({})
+                .buildService({ create: methods.create })
+                .buildService({ detail: methods.detail })
+                .buildService({ update: methods.update })
+                .buildService({ delete: methods.delete })
+                .buildService({ list: methods.list }).createService;
+            const service = serviceFactory({});
+            await expect(service.detailHandler()(id, context)).resolves;
```

Comment:
Why is needed the `await exect(...).resolves`? It the promise is rejected, await throws an error which automatically bubbles up and will get caught by jest, leading to test case failure on that error.